### ISSUE TITLE
AWSShellScript fails when we rotate the keys of an existing service connection

### ISSUE THREAD

**Issue by jbm00n** [aws/aws-toolkit-azure-devops#428]
**Description**

We use shared service connections and all works fine. We now want to rotate the key from AWS, but after we update the key access and secret keys, the deployment fails with error:

##[error]InvalidClientTokenId: The security token included in the request is invalid

After trying with multiple new different keys (all failed), we created a new service connection, with same keys, and this one worked.

**To reproduce**

- Create a service connection
- Use it and verifies it worked
- Generate new AWS keys
- Edit the service connection with the new keys
- Retry to deploy

**Environment**

-  Cloud based
-   AWS Toolkit for Azure DevOps version: 1.11

**Comment by bryceitoc9** [comment]
Hmm, I can't seem to replicate this, but I might have too shallow of a test. Could you elaborate on what you're doing? Is the new set of credentials from the same account? Do the credentials come from a different account which doesn't have the right permissions? Also, if you're migrating from access/secret key to an assumed role with a session token, are you setting the token correctly? Also, is the token valid and not expired?

My test specifically is a brand new job running an `aws s3 ls` in an AWS Shell Script, and I'm rotating the access/secret/session token with a new one from the same account; the job still successfully lists buckets after the rotation.

**Comment by jbm00n** [comment]
Thank you for your time. It's the same account, same IAM user, just generated new access key.

The service connection in DevOps is the same one, I did not touch the config, only updated the access and secret keys.

I haven't tried to reproduce though with a fresh service connection. So currently I have 2 service connections:
- Old one with new keys -> Failed
- New one with same new keys -> Works

I contacted Azure support, but of course they sent me back here as the task is implemented by AWS. I was just hoping there would be a way to debug why the original service connection cannot handle the new keys.

**Comment by bryceitoc9** [comment]
I wonder, do you have any other settings in the old service connection that you're not overwriting, e.g. role to assume or session token (the error message seems to suggest this is tied to a token)? I don't think Azure DevOps will let you zero out one of these fields, unfortunately.

**Comment by jbm00n** [comment]
Yes, that's what I am wondering as well. The role is the same, so I am not touching that. But the session token is of course something I do not populate. And as you mentioned I can't zero it out, when I click on the field (which is masked) it's just a blank field, but I am wondering if in the back it actually maintains the older value.

**Comment by jbm00n** [comment]
In Azure DevOps, is the service connection for AWS maintained by Azure or is this something AWS team provides? Trying to figure who I could ask to investigate how the session token is being maintained

**Comment by kunal-stagepilot** [comment]
Thank you @jbm00n!!
I've been trying to find a resolution for this issue for the past couple of days.

**Comment by kunal-stagepilot** [comment]
I created an IAM User with programmatic access for deploy purposes. This user's credentials doesn't have a session token generated.
I had previously used my personal user's credentials which included a session token.

When I switched back to the IAM user credentials created for deploy purposes in for AWS Service Connection, it was still passing the session token for my personal user. Following the steps mentioned by @jbm00n resolved the issue.

This error was being logged in Release pipeline:

> Test for existence of <my stack name> returned error: 'InvalidClientTokenId: The security token included in the request is invalid'
> ##[error]Error: Error uploading template file: InvalidToken: The provided token is malformed or otherwise invalid.
